wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -37,8 +37,9 @@ sealed abstract class Wallet extends LockedWallet with UnlockedWalletApi {
       feeRate: FeeUnit,
       fromAccount: AccountDb): Future[Transaction] = {
     require(
-      address.networkParameters == networkParameters,
-      s"Cannot send to address on other network, got ${address.networkParameters}")
+      address.networkParameters.p2pkhNetworkByte == networkParameters.p2pkhNetworkByte,
```

Comment:
isn't this only going to work for `p2pkh`? Wont' we need this for `p2sh` and `segwit` as well?

Reply:
The reason for using `p2pkhNetworkByte` is because on testnet and regtest they are equal. When parse an address ie `2N4WCUEPY3Y52PVBn5Wgz5c1RoMqmbUGXri`, it will be parsed as a testnet address but it is also valid on regtest. So if we were just comparing `networkParameters` then it'd be marked invalid even though it is not. So, we compare `p2pkhNetworkByte` because they are equal across testnet and regtest. We could have used `p2sh` or `segwit` but it doesn't really matter, we are just trying to compare the `networkParameters` between if it is valid on mainnet or test chains(testnet and regtest)

Reply:
So why don't we just make a nice helper method inside if `TestNet3` and `RegTest` that says `isSameNetworkBytes()` or something?

Reply:
That is a good idea